### PR TITLE
Added Regression to the mean oversight

### PR DESCRIPTION
**Pull request by bhagyakjain**
Changes -

- Added the Regression to the mean oversight in the oversights intents/oversights folder
- Added a new manually created dataset to test the similarity between the ranks method

### CODE REVIEW

File: intents/oversights/regression_to_mean.py [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+oversight suggestion is passed.
+"""
+
+import sys
+
+import datetime
+
+import math
+
+sys.path.append("../..")
```
Comment (arijitde-goog) [inline intents/oversights/regression_to_mean.py:32]: This is a little weird. Can we add __init__.py in our folders to make them packages, so that they automatically can be imported?

File: intents/test_topk.py [hunk 2]
```diff
@@ -183,7 +185,26 @@ def test_7():
...
+                             date_range=('23/05/2010', '25/05/2011'),
+                             date_column_name='date',
+                             date_format='%d/%m/%Y')
+    print(query_result)
+    expected_result = """  User Name  Rating
+0      Benq    3400
+1     300iq    4300
+2       cba    5200
+3   tourist    6100"""
+    assert(expected_result == query_result[0].to_string())
```
Comment (arijitde-goog) [inline intents/test_topk.py:207]: Did you mean to assert the query_result[1] here as well?

File: intents/util/constants.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+limitations under the License.
+"""
+
+"""
+This module stores the cuttoff of different that are fixed while
+execution of the code.
+These may be changed to decrease the number of false positives.
+"""
+# cut-off of set intersection method in regression to the mean
+RTM_SET_INTERSECTION = .7
```
Comment (arijitde-goog) [inline intents/util/constants.py:22]: RTM_SET_INTERSECTION_CUT_OFF

File: intents/util/constants.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+# cut-off of similarity in ranks method in regression to the mean
+RTM_RANK_VECTOR = 30
```
Comment (arijitde-goog) [inline intents/util/constants.py:24]: RTM_RANK_VECTOR_SIMILARITY_THRESHOLD

File: intents/util/constants.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+"""
+This module stores the cuttoff of different that are fixed while
+execution of the code.
```
Comment (arijitde-goog) [inline intents/util/constants.py:18]: Reword this to be a little more generic. "stores various constants that affect behavior of intents / oversights"

**Comment by bhagyakjain** [comment]
I am not sure if I have added __init__.py in the way it is supposed to be added, but the imports works & test cases pass.

File: intents/oversights/regression_to_mean.py [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+Regression toward the mean is the phenomenon that arises if a random
+variable is extreme on its first or first few measurements but closer
+to the mean or average on further measurements.
+If the top-k is queried for a time window in which the results look much
+different from the previous window, or the next window of the same size, the
+oversight suggestion is passed.
+"""
+
+import sys
+
```
Comment (arijitde-goog) [inline intents/oversights/regression_to_mean.py:27]: remove blank lines between the imports

File: intents/topk.py [hunk 7]
```diff
@@ -20,9 +20,90 @@
 Some of the operations are optional.
 """
 
+from oversights.regression_to_mean import regression_to_mean
+
```
Comment (arijitde-goog) [inline intents/topk.py:24]: same here
